samples/csharp_dotnetcore/50.teams-messaging-extensions-search/Bots/TeamsMessagingExtensionsSearchBot.cs

```diff
@@ -99,5 +124,125 @@ private async Task<IEnumerable<(string, string, string, string, string)>> FindPa
+            var previewcard = new ThumbnailCard
+            {
+                Title = "Adaptive Card",
+                Text = "Please select to get Adaptive card"
+            };
+
+            var adaptiveList = FetchAdaptive(filepath);
+            var attachment = new MessagingExtensionAttachment
+            {
+                ContentType = AdaptiveCards.AdaptiveCard.ContentType,
+                Content = adaptiveList.Content,
+                Preview = previewcard.ToAttachment()
+            };
+
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "result",
+                    AttachmentLayout = "list",
+                    Attachments = new List<MessagingExtensionAttachment> { attachment }
+                }
+            };
+        }
+
+        public MessagingExtensionResponse GetConnectorCard()
+        {
+            string filepath = "./Resources/connectorCard.json";
+            var previewcard = new ThumbnailCard
+            {
+                Title = "O365 Connector Card",
+                Text = "Please select to get Connector card"
+            };
+
+            var connector = FetchConnector(filepath);
+            var attachment = new MessagingExtensionAttachment
+            {
+                ContentType = O365ConnectorCard.ContentType,
+                Content = connector.Content,
+                Preview = previewcard.ToAttachment()
+            };
+
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "result",
+                    AttachmentLayout = "list",
+                    Attachments = new List<MessagingExtensionAttachment> { attachment }
+                }
+            };
+        }
+
+        public static Attachment FetchAdaptive(string filepath)
+        {
+            var adaptiveCardJson = File.ReadAllText(filepath);
+            var adaptiveCardAttachment = new Attachment
+            {
+                ContentType = AdaptiveCard.ContentType,
+                Content = JsonConvert.DeserializeObject(adaptiveCardJson)               
+            };
+
+            return adaptiveCardAttachment;           
+        }
+
+        public Attachment FetchConnector(string filepath)
+        {
+            var connectorCardJson = File.ReadAllText(filepath);
+            var connectorCardAttachment = new MessagingExtensionAttachment
+            {
+                ContentType = O365ConnectorCard.ContentType,
+                Content = JsonConvert.DeserializeObject(connectorCardJson)
+            };
+
+            return connectorCardAttachment;
+        }
+
+        public MessagingExtensionResponse GetResultGrid()
+        {
+            List<string> images = new List<string>();
+            images.Add("https://picsum.photos/300?image=882");
```

Comment:
This range of images from so many sources seems strange.  Can we include these in the solution, and reference them using baseUrl in appsettings.json?  Or, from their github resource path?

Reply:
Sure done this change as well